src/libasr/ASR.asdl

```diff
@@ -259,7 +259,8 @@ expr
     | ComplexRe(expr arg, ttype type, expr? value)
     | ComplexIm(expr arg, ttype type, expr? value)
     | DictItem(symbol a, expr key, expr? default, ttype type)
-
+    | ListItem(symbol a, expr pos, ttype type, expr? value)
+    | ListSection(symbol a, array_index section, ttype type, expr? value)
```

Comment:
I wonder if this should be an `expr` instead of `symbol`:
```suggestion
    | ListSection(expr a, array_index section, ttype type, expr? value)
```
So that we can represent things like `(a+b)[2:3]` or indexing/slicing of the result of a function such as `f()[2:3]`.

I think we can do this improvement in subsequent PRs, not here.

Reply:
I opened up an issue for this at #475.

Reply:
Sure, will do that